test/show.jl

```diff
@@ -1006,3 +1001,17 @@ end
     s = String(take!(io))
     @test contains(s, " in Base.Math ")
 end
+
+@testset "typeinfo" begin
+    @test replstr([[Int16(1)]]) == "1-element Array{Array{Int16,1},1}:\n [1]"
+    @test showstr([[Int16(1)]]) == "Array{Int16,1}[[1]]"
+    @test showstr(Set([[Int16(1)]])) == "Set(Array{Int16,1}[[1]])"
+    @test showstr([Float16(1)]) == "Float16[1.0]"
+    @test showstr([[Float16(1)]]) == "Array{Float16}[[1.0]]"
+    @testset "nested Any eltype" begin
```

Comment:
Would be nice to test with other abstract types like `Real`.

Reply:
The specifity of `Any` (compared to an abstract type in general) is that a collection is `isa Any` and its elements are also `isa Any`, and I don't know which other type would satisfy this. But I can of course add a test for `Real`.

Reply:
OK. I was thinking about the `Any` special case (see [my comment](https://github.com/JuliaLang/julia/pull/24651/files/03ef631cc314b2f894c03c35b9b181273bdc1eaa..bfac2906bc42a7b1f5377ee944b689982be3f6e9#r152790948)).